### PR TITLE
Add functionality to manage plugins

### PR DESCRIPTION
**Pull request by matthewnolf**
Adds functions to get, install and uninstall plugins. 

The `Plugin` struct is not currently exhaustive with all the fields returned by the API - I didn't see a use case for the others but can be added if required.

One question I have is that: these functions interact with Grafana-Com API, not Grafana itself. Should that be called out explicitly, or belongs somewhere else? Comparing this to the stack functionality, there is no such callout, but perhaps that is less ambiguous.

### CODE REVIEW

**Comment by julienduchesne** [comment]
> One question I have is that: these functions interact with Grafana-Com API, not Grafana itself. Should that be called out explicitly, or belongs somewhere else? Comparing this to the stack functionality, there is no such callout, but perhaps that is less ambiguous.

Prefix everything with `Cloud`. It wasn't done for the stacks but it should have

**Comment by julienduchesne** [comment]
Since the test framework is not very useful in this repo, I'll wait until the Terraform provider PR is up as well before reviewing + merging this one. We use the Terraform provider as the test framework. You can `go get` this branch's changeset to work with the provider while this isn't merged

**Comment by matthewnolf** [comment]
> Since the test framework is not very useful in this repo, I'll wait until the Terraform provider PR is up as well before reviewing + merging this one. We use the Terraform provider as the test framework. You can `go get` this branch's changeset to work with the provider while this isn't merged

Makes sense to me, will have the provider PR up shortly and will link here

File: plugin.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
+package gapi
```
Comment (julienduchesne) [inline plugin.go]: I'd also call the file `cloud_plugin.go`

**Review by julienduchesne — APPROVED** [review]

**Pull request merged**